src/app/network/upload.ts

```diff
@@ -130,34 +128,47 @@ export async function uploadFile(bucketId: string, params: IUploadParams): Promi
 
   params.abortController?.signal.addEventListener('abort', onAbort);
 
-  if (useMultipart) {
-    uploadPromise = facade.uploadMultipart(bucketId, params.mnemonic, file, {
-      uploadingCallback: params.progressCallback,
-      abortController: uploadAbortController,
-      parts: Math.ceil(params.filesize / partSize),
-      continueUploadOptions: params?.continueUploadOptions,
-    });
-  } else {
-    uploadPromise = facade.upload(bucketId, params.mnemonic, file, {
-      uploadingCallback: params.progressCallback,
-      abortController: uploadAbortController,
-      continueUploadOptions: params?.continueUploadOptions,
-    });
-  }
-
-  return uploadPromise
-    .catch((err: ErrorWithContext) => {
-      if (connectionLost) {
-        throw new ConnectionLostError();
+  async function retryUpload(): Promise<string> {
+    const MAX_TRIES = 3;
+    const RETRY_DELAY = 1000;
+    let uploadPromise: Promise<string>;
+
+    for (let attempt = 1; attempt <= MAX_TRIES; attempt++) {
+      try {
+        if (useMultipart) {
+          uploadPromise = facade.uploadMultipart(bucketId, params.mnemonic, file, {
+            uploadingCallback: params.progressCallback,
+            abortController: uploadAbortController,
+            parts: Math.ceil(params.filesize / partSize),
+            continueUploadOptions: params?.continueUploadOptions,
+          });
+        } else {
+          uploadPromise = facade.upload(bucketId, params.mnemonic, file, {
+            uploadingCallback: params.progressCallback,
+            abortController: uploadAbortController,
+            continueUploadOptions: params?.continueUploadOptions,
+          });
+        }
+
+        return await uploadPromise;
+      } catch (err) {
+        if (connectionLost) {
+          throw new ConnectionLostError();
+        }
+
+        console.warn(`Attempt ${attempt} of ${MAX_TRIES} failed:`, err);
+        Sentry.captureException(err, { extra: (err as ErrorWithContext).context });
+
+        if (attempt === MAX_TRIES) {
+          throw err; // ❌ Last try failed
```

Comment:
If you do this one: 

```
const lastTryFailed = attempt === MAX_TRIES

if (lastTryFailed) {
   throw err;
}
```

Then the code is self-explanatory and the comment is not needed, which indeed is the best way to prevent outdated comments, as comments do not fail if they are wrong, the code does.